### PR TITLE
String to slug

### PR DESCRIPTION
**Pull request by FrankKair**
Creates a method to return a slug version of a string.
Example:

```swift
let str = "  A nice & hög _ Str    "
str.toSlug() // Returns "a-nice-&-hog-str"
```

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a changelog entry describing my changes.

### CODE REVIEW

**Comment by FrankKair** [comment]
Do you guys think I'm missing some edge case here?
Or... is there a more efficient way of doing this?

File: Sources/Extensions/SwiftStdlib/StringExtensions.swift [hunk 1]
```diff
@@ -425,6 +425,34 @@ public extension String {
 		// https://stackoverflow.com/questions/42822838
 		return words().count
 	}
+
+    /// SwifterSwift: Transforms the string into a slug string.
+    ///
+    ///        "Swift is amazing".toSlug() -> "swift-is-amazing"
+    ///
+    /// - Returns: The string in slug format.
+    public func toSlug() -> String {
```
Comment (SD10) [inline Sources/Extensions/SwiftStdlib/StringExtensions.swift]: I've never heard of a "slug" before 😂 But if this is something people need then it looks 👌

Comment (wdcurry) [reply Sources/Extensions/SwiftStdlib/StringExtensions.swift]: Slugs are used (in my experience) as safe endpoints to urls for blog posts etc.. Wordpress is all about the slug.

Comment (SD10) [reply Sources/Extensions/SwiftStdlib/StringExtensions.swift]: Ahhh, I see. Thanks for the info @wdcurry 👍

**Comment by WestFlow127** [comment]
func toSlug() -> String does not remove all emoji's for example "str.toSlug() : test-🙋🏻‍♀️" ... I am using this function to create safe endpoints for our app. Would be nice to see an update to this issue in the next release. Thanks

**Pull request merged**